### ISSUE TITLE
Flip the Object and Context Responses around

### ISSUE THREAD

**Issue by terracoda** [phetsims/ratio-and-proportion#245]
As discussed in the design meeting on Tuesday, we want to try and flip the object responses and context responses around for the hand interactions.

Reason, we like the order of the information when Lock Ratio check box is checked , and in that case, proximity to ratio is the value for the object responses.

I've sketched out an example [CWT for the Both Hands interaction in this table](https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#heading=h.j01dg7fai4u8) in the design document.

The sliders will be similar, and in the case of the sliders there are also "distance progress indicators".

A few issues ago, I took the progress indicators out of the both hands interaction, but with the information flipped around, it might make sense to put them back. @zepumph, please do whatever is most straight forward, and then I can test and we can discuss.

**Comment by zepumph** [comment]
@terracoda asked me not to do this until she could have a listen to some of the other changes I made in the description. @terracoda let me know when we are ready for this.

**Comment by terracoda** [comment]
@zepumph, I made a table with a [small test of the Left hand slider with the design flipped](https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#heading=h.hq2zvuiwvq6f)

I think it is safe to flip the design.

**Comment by terracoda** [comment]
My test was applying the new logic, but we can discuss the logic before or after the flip - whatever makes sense to you.

**Comment by terracoda** [comment]
Oops, forgot to re-assign to you, @zepumph.

When you do this, I see 3 parts to the flip:
1. Both Hands - looks like a direct flip since the ratio responses are currently on their own in context responses
2. Sliders - not a direct flip, since ratio responses are combined with distance (and progress) in context responses
3. Adjustment of the current logic that replaces context responses with distance progress responses

Do you see it this way as well @zepumph?

**Comment by terracoda** [comment]
@zepumph, I'll create separate issues numbers 2 and 3 above, in case that is helpful.
This issue can be for number 1.

**Comment by zepumph** [comment]
This should be implemented, please review @terracoda.

**Comment by zepumph** [comment]
When you review this, know that all flipping logic from side-issues (#261 and #262) has also been implemented.

**Comment by terracoda** [comment]
With my challenge 1:3, the distance-progress response are reversed, just going to test if is always the case they are reversed.

- I move hands up, sims says "closer together"
- I move down, sim says "farther apart"

Just double checking now.

**Comment by terracoda** [comment]
I will do a fresh pull first.

**Comment by terracoda** [comment]
The issue is that, when ratio is locked, and I move the hands down, I get "farther apart", and when i move them up, I get "closer together".

**Comment by terracoda** [comment]
**with ratio locked, Moving hands down**
![Screen Shot 2020-12-07 at 6 07 33 PM](https://user-images.githubusercontent.com/1902116/101408531-1c4e5b80-38b7-11eb-8a84-038209967287.png)

**with ratio locked, Moving hands up**
![Screen Shot 2020-12-07 at 6 06 40 PM](https://user-images.githubusercontent.com/1902116/101408448-fc1e9c80-38b6-11eb-8787-835febca4629.png)

**Comment by terracoda** [comment]
@zepumph, I don't seem to get distance-progress if I have focus on a slider and ratio is locked. Here are three consecutive steps going up in the same distance region.

![Screen Shot 2020-12-07 at 6 11 52 PM](https://user-images.githubusercontent.com/1902116/101409055-e198f300-38b7-11eb-9b1d-6cac6e7ade74.png)
![Screen Shot 2020-12-07 at 6 12 09 PM](https://user-images.githubusercontent.com/1902116/101409085-ebbaf180-38b7-11eb-8757-97f1a2cdee07.png)
![Screen Shot 2020-12-07 at 6 12 21 PM](https://user-images.githubusercontent.com/1902116/101409094-ee1d4b80-38b7-11eb-8b42-a82e78b05a09.png)

**Comment by terracoda** [comment]
Last comment is more appropriate for issue #262, maybe.

**Comment by terracoda** [comment]
Oh, last 2 comments, https://github.com/phetsims/ratio-and-proportion/issues/245#issuecomment-740198448 and https://github.com/phetsims/ratio-and-proportion/issues/245#issuecomment-740195818

maybe these comments should go in #262?
Anyways, this issue is connect to #261 and #262

**Comment by zepumph** [comment]
From the design meeting today, @terracoda reiterated two bugs with the current implementation:

* The distance-progress isn't working when ratio is locked
* When using both-hands, the distance-progress is often (always!?!) backwards.

**Comment by zepumph** [comment]
>The distance-progress isn't working when ratio is locked

This was fixed in https://github.com/phetsims/ratio-and-proportion/commit/65abe5698dbebd6c307309ed5a2cb5a77324fa41 by making sure that we only calculate the distance changed when we create the description. Before it was in a listener and could change multiple times before needing a value of "closer" or "farther", and it would be stale. This should be fixed now.

**Comment by zepumph** [comment]
@terracoda, I believe that both bugs are fixed. I at least think it is ready for another review. Good luck!

**Comment by terracoda** [comment]
Ok, cool.

**Comment by terracoda** [comment]
- closer/farther distance progress responses is no working for Both Hands with Lock Ratio checked.
- with the hands unlocked, the Both Hands is getting distance progress responses

**Comment by terracoda** [comment]
I still need to check with the tick marks, but don't anticipate issues.

**Comment by terracoda** [comment]
- the distance progress is working with the tick marks on.

With ratio locked, it seems when the left hand is visually "on a tick mark", it described as being "around" to tick mark.

Here's 2 screen shots. Not sure if this is directly related to this issue, but since I am testing this issue, I just need to post it here before I lose my train of thought:
![Screen Shot 2020-12-09 at 4 15 54 PM](https://user-images.githubusercontent.com/1902116/101680110-cc000680-3a3a-11eb-9002-12e60fd45b0a.png)
![Screen Shot 2020-12-09 at 4 19 38 PM](https://user-images.githubusercontent.com/1902116/101680119-cefaf700-3a3a-11eb-95e5-3f1047147382.png)

**Comment by terracoda** [comment]
Double checking with Lock Ratio unchecked.
Ok, this seems to happen only with Lock Ratio checked.

I can start out with Both Hands at 1:3 with each on the correct tick mark, but by the time the Left Hand gets to Tick Mark 2, the descriptions says "around tick mark 2" with right hand "on tick mark 6".

Also, if I start out at 1:3 and then move the hands down as far as they can go with ratio locked, and then move back up again, the left hand is described as "around tick mark 1" when it visually gets back on 1.

**Comment by terracoda** [comment]
I commented over in https://github.com/phetsims/ratio-and-proportion/issues/209#issuecomment-742020700 just in case the above comments should be in that issue.

**Comment by terracoda** [comment]
@zepumph, in principle the correct object and context responses are firing for Both Hands. 
The actual position of the left hand can become askew when ratio is locked.

Shall we work on that more in issue #209?
Assigning to for closing.

**Comment by zepumph** [comment]
I created https://github.com/phetsims/ratio-and-proportion/issues/279 for that bug. It sounds like this issue about flipping responses is done. Closing
